src/publisher/middleware.py

```diff
@@ -2,19 +2,13 @@
 import json
 import logging
 from .utils import isint
-from rest_framework.response import Response as RESTResponse
-from .api_v2_views import flatten_accept, ErrorResponse, _ctype
-from django.http import HttpResponse
+from .api_v2_views import flatten_accept, _ctype, http_406
 
 LOG = logging.getLogger(__name__)
 
 
 def get_content_type(resp):
```

Comment:
can our custom `HttpResponse` obj in views replace this?

Reply:
it can, but middleware needs to be more robust that the view code, responses passing through them may not always be what we can control.